aten/src/ATen/native/native_functions.yaml

```diff
@@ -230,6 +230,8 @@
 
 - func: slice(Tensor self, int64_t start=0, int64_t end=9223372036854775807, int64_t step=1, int64_t dim=0) -> Tensor
 
+- func: slice_dim(Tensor self, int64_t dim, int64_t start=0, int64_t end=9223372036854775807, int64_t step=1) -> Tensor
```

Comment:
Let's add this to `SKIP_PYTHON_BINDINGS` in `gen_variable_type.py` since it doesn't seem useful in Python. (I'd rather people stick to the x[...] syntax)

Reply:
BTW why do we need `slice_dim` in C++?

Reply:
It's to avoid having to specify end when you just want to slice to the end; you either need to specify 9223372036854775807 because we couldn't have self.size() as the default in C++ and often self is a result of another computation, so not trivial to just do the equivalent at the call site.

we should maybe just change the order of arguments in slice (usually we put dim at the end, so that's why it was there originally); I don't really have a strong opinion either way.

Reply:
I'm not sure it's worth adding `slice_dim` as a public C++ function/method either. We already have `narrow` and `slice`.  

For example,

```
input.slice_dim(dim, k + 1)
```

Can be written as:

```
input.slice(k + 1, input.size(dim), 1, dim)
```

Or equivalently:

```
input.narrow(dim, k + 1, input.size(dim) - k - 1)
```

Or `slice_dim` can just be defined locally in Functions.cpp or as a lambda in `prod_safe_zeros_backward`.

Reply:
Can we just change the order of parameters in `slice` to match `slice_dim`? `slice_dim` order ends up cleaner (you can only get `end` wrong) and I'd argue it's closer to the python API anyway, i.e.:
`input.slice_dim(dim, k+1)` is the same order as `tensor[(slice(None),)*dim+(slice(k+1),)]`

Reply:
Also most of our other functions that take a dim take it first.

Reply:
OK